compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -8086,7 +8086,13 @@ public void visit(BLangStatementExpression bLangStatementExpression) {
     public void visit(BLangQueryExpr queryExpr) {
         boolean prevIsVisitingQuery = this.isVisitingQuery;
         this.isVisitingQuery = true;
-        BLangStatementExpression stmtExpr = queryDesugar.desugar(queryExpr, env, getVisibleXMLNSStmts(env));
+        BLangStatementExpression stmtExpr;
+        if (queryExpr.stmtExpr == null) {
+            stmtExpr = queryDesugar.desugar(queryExpr, env, getVisibleXMLNSStmts(env));
+            queryExpr.stmtExpr = stmtExpr;
+        } else {
+            stmtExpr = queryExpr.stmtExpr;
```

Comment:
We assume that we have desugared here but will this assumption be true always ?

Reply:
I have revert this and updated the logic 35169c467dc78f97ec3a6633f3fe9ded0aecd86c